### DISCUSSION TITLE
RFE: Deduplicate data among binary packages produced by the same source package in the rpmdb

### DISCUSSION THREAD

**Discussion by Conan-Kudo** [rpm-software-management/rpm#2829] [Ideas]
With more and more packages, especially in the distributions that do library subpackaging by soname (such as openSUSE, Mageia, and OpenMandriva), there's a lot of data that comes from the source package that has duplicate entries in the rpmdb per binary package. The largest offender of this is the changelog data, but sometimes also license tags and a few other things wind up being the same.

A nice optimization for this would be to have a skeleton "source package" rpmdb entry with that data and put the common data there and have the binary package entries in the rpmdb link to that. This can lead to significant shrinkage of the rpmdb depending on distro policy on packaging.

**Comment by kloczek** [comment]
Wate of time.
Deduplication it is something which should be done on fs layer.

**Comment by pmatilai** [comment]
All that data is in the header, which you cannot split without breaking signatures and digests. Which you want to be able to verify even for installed packages too.

As in, this is not solvable without fundamental redesign of the rpmdb.
